cli/tests/keyChange.test.ts

```diff
@@ -0,0 +1,326 @@
+import { isArm } from "maci-circuits";
+import {
+  deploy,
+  deployPoll,
+  deployVkRegistryContract,
+  genProofs,
+  mergeMessages,
+  mergeSignups,
+  proveOnChain,
+  publish,
+  setVerifyingKeys,
+  signup,
+  timeTravel,
+  verify,
+} from "../ts/commands";
+import {
+  INT_STATE_TREE_DEPTH,
+  MSG_BATCH_DEPTH,
+  MSG_TREE_DEPTH,
+  STATE_TREE_DEPTH,
+  VOTE_OPTION_TREE_DEPTH,
+  coordinatorPrivKey,
+  coordinatorPubKey,
+  processMessageTestZkeyPath,
+  tallyVotesTestZkeyPath,
+  testProcessMessagesWasmPath,
+  testProcessMessagesWitnessPath,
+  testProofsDirPath,
+  testRapidsnarkPath,
+  testTallyFilePath,
+  testTallyVotesWasmPath,
+  testTallyVotesWitnessPath,
+} from "./constants";
+import { Keypair } from "maci-domainobjs";
+import { cleanVanilla } from "./utils";
+import { readFileSync } from "fs";
+import { expect } from "chai";
+import { genRandomSalt } from "maci-crypto";
+import { DeployedContracts } from "../ts";
+
+describe("keyChange tests", function () {
+  const useWasm = isArm();
+  this.timeout(900000);
+
+  let maciAddresses: DeployedContracts;
+
+  // before all tests we deploy the vk registry contract and set the verifying keys
+  before(async () => {
+    // we deploy the vk registry contract
+    await deployVkRegistryContract(true);
+    // we set the verifying keys
+    await setVerifyingKeys(
+      STATE_TREE_DEPTH,
+      INT_STATE_TREE_DEPTH,
+      MSG_TREE_DEPTH,
+      VOTE_OPTION_TREE_DEPTH,
+      MSG_BATCH_DEPTH,
+      processMessageTestZkeyPath,
+      tallyVotesTestZkeyPath,
+    );
+  });
+
+  describe("keyChange and new vote (new vote has same nonce)", () => {
+    after(async () => {
+      cleanVanilla();
+    });
+    const keypair1 = new Keypair();
+    const keypair2 = new Keypair();
+    const initialNonce = 1;
+    const initialVoteOption = 0;
+    const initialVoteAmount = 9;
+    const pollId = 0;
+    let stateIndex = 0;
+    const expectedTally = initialVoteAmount - 1;
+    const expectedPerVOteOptionTally = (initialVoteAmount - 1) ** 2;
+
+    before(async () => {
+      // deploy the smart contracts
+      maciAddresses = await deploy(STATE_TREE_DEPTH);
+      // deploy a poll contract
+      await deployPoll(
+        90,
+        25,
+        25,
+        INT_STATE_TREE_DEPTH,
+        MSG_BATCH_DEPTH,
+        MSG_TREE_DEPTH,
+        VOTE_OPTION_TREE_DEPTH,
+        coordinatorPubKey,
+      );
+      stateIndex = parseInt(await signup(keypair1.pubKey.serialize()));
```

Comment:
```suggestion
      stateIndex = await signup(keypair1.pubKey.serialize());
```

nit: it's better to return `number` type from `signup()`

Reply:
Can you please dismiss your change request as this is something not related to this PR? will amend in another PR